rpc/src/jsonrpc.rs

```diff
@@ -30,3 +31,17 @@ pub struct Pool {
     /// the used capacity in bytes
     pub used: u64,
 }
+use std::str::FromStr;
+impl FromStr for ShareProtocol {
+    type Err = ();
+
+    fn from_str(s: &str) -> Result<ShareProtocol, ()> {
+        match s.to_lowercase().trim() {
+            "none" => Ok(ShareProtocol::None),
+            "nvmf" => Ok(ShareProtocol::Nvmf),
+            "iscsi" => Ok(ShareProtocol::Iscsi),
+            "nbd" => Ok(ShareProtocol::Nbd),
+            _ => Err(()),
```

Comment:
Is the error message eventually printed to the user indicative? i.e. something like invalid parameter with string value.

Reply:
Good point... I'll amend.

Reply:
Error message added in updated code.